### ISSUE TITLE
Logback's XML configuration for the SentryAppender is not allowing dsn as option

### ISSUE THREAD

**Issue by bogdanrusutradeshift** [getsentry/sentry-java#1052]
## Important Details

How are you running Sentry?

On-Premise w/ Docker, version 3.1.1

## Description

I followed the example from https://docs.sentry.io/platforms/java/guides/logback/#dsn-configuration to add a Sentry appender and put the DSN value directly in logback.xml file. I explicitly removed the dns from sentry.properties to test if the XML config works as the docs are saying but it seems it's not working. In the container logs I see:
`ERROR in ch.qos.logback.core.joran.spi.Interpreter@147:14 - no applicable action for [dsn], current ElementPath is [[configuration][appender][dsn]]`

In the class itself it seems that `options` is present and passed to the init method, but the XML parsing might have an issue.

## Steps to Reproduce

1. Setup the On-Premise w/ Docker
2. Test with sentry-cli using the DNS from local Sentry
3. Add Logback Sentry's Appender using the XML configuration including the `dsn` property 
4. Start application and check if the Appender has been initialized

### What you expected to happen

Appender is initialized using the dns from XML.

### Possible Solution

Correct the documentation.

**Comment by maciejwalkowiak** [comment]
@bogdanrusutradeshift Could you please post your logback.xml file here? `current ElementPath is [[configuration][appender][dsn]]` suggests that you skipped `options` but I may be missing something.

**Comment by bogdanrusutradeshift** [comment]
> @bogdanrusutradeshift Could you please post your logback.xml file here? `current ElementPath is [[configuration][appender][dsn]]` suggests that you skipped `options` but I may be missing something.

Hi @maciejwalkowiak ! The above error is complaining about the `dsn` without the `options` but I get the same if I place `dsn` inside an `options` tag, forgot to add that line also (sorry about the confusion). It was clear after checking the SentryAppender class that I had to have `options.dsn` but that didn't work either, the error included a similar message as above. 
I also had the `dsn` configured in `sentry.properties` but I don't see how might interfere with this.
Let me know if you want me to add more details.

**Comment by maciejwalkowiak** [comment]
@bogdanrusutradeshift We have a sample project that sets the DSN through Logback XML and it's working fine https://github.com/getsentry/sentry-java/tree/main/sentry-samples/sentry-samples-logback so it would be great if you could provide a sample that reproduces an issue.

**Comment by bogdanrusutradeshift** [comment]
@maciejwalkowiak After some more testing I figure out that the setup I had actually used an older version (1.7.24) and hence the issues.

I've also debugged and indeed, the `options.dsn` property is correctly picked up when the `SentryAppender` for Logback is first initialized.

Thanks for all the support in this thread and also on [SO](https://stackoverflow.com/questions/64879197/add-a-hook-to-sentry-logback-to-scrub-data/64907852)!

This ticket can be closed now.
